deliver/lib/deliver/app_screenshot.rb

```diff
@@ -293,13 +302,27 @@ def self.devices
       }
     end
 
+    def self.resolve_ipadpro_conflict_if_needed(device_type, filename)
+      is_3rd_gen = filename.include?("(3rd generation)")
+      if is_3rd_gen
+        if device_type == ScreenSize::IOS_IPAD_PRO
+          return ScreenSize::IOS_IPAD_PRO_12_9
+        elsif device_type == ScreenSize::IOS_IPAD_PRO_MESSAGES
+          return ScreenSize::IOS_IPAD_PRO_12_9_MESSAGES
+        end
+      end
+      device_type
+    end
+
     def self.calculate_screen_size(path)
       size = FastImage.size(path)
 
       UI.user_error!("Could not find or parse file at path '#{path}'") if size.nil? || size.count == 0
 
       # Walk up two directories and test if we need to handle a platform that doesn't support landscape
-      path_component = Pathname.new(path).each_filename.to_a[-3]
+      path_filenames = Pathname.new(path).each_filename.to_a
+      path_component = path_filenames[-3]
+      filename = path_filenames.to_a[-1]
```

Comment:
Lets replace this strangeness by https://ruby-doc.org/stdlib-2.1.3/libdoc/pathname/rdoc/Pathname.html#method-i-basename, possibly before the `devices.each` in line 333 (and replace this here with what it was before)